lib/systems/examples.nix

```diff
@@ -356,6 +356,12 @@ rec {
     useLLVM = true;
   };
 
+  wasm32-unknown-none = {
+    config = "wasm32-unknown-none-unknown";
+    rust.rustcTarget = "wasm32-unknown-unknown";
+    useLLVM = true;
+  };
```

Comment:
Is this useful for anything besides Rust?

Reply:
Maybe? Idk of enough languages that can compile to WASI-less wasm.

Reply:
Where did the triple come from then?

Anyway, if it's Rust-only, I'm wondering whether pkgsCross is the right mechanism.  We can already handle (buildRustPackage) packages containing wasm32-unknown-unknown builds (and we can't use pkgsCross for those, because they contain a mix of wasm and native code).  There's no standard library to build, nor (if it's only Rust) are there likely to be packaged libraries to link against.  So why not just let buildRustCrate take a target parameter (if it doesn't already)?  That would also be useful for other custom targets, as well as the -none ones.

@Ericson2314 wdyt?

Reply:
I think the way buildRustCrate and crate2nix work is slightly more involved than that, it's not just about passing `--target` to rustc, there's a whole logic of matching dependencies depending on the target's family, arch etc. Crate2nix directly uses the nixpkgs cross compilation logic. Also what you are commenting on is just about having an example of it. I made the logic pretty generic so it will apply to other wasm targets that rust support (haven't tested tho!).

Reply:
I can certainly see a future where we might want to statically link together some rust crates with some static C code (maybe [via Emscripten](https://developer.mozilla.org/en-US/docs/WebAssembly/existing_C_to_Wasm), and using the nixpkgs cross ecosystem should be the right vehicle for it, no?

Reply:
Well that's the thing — Emscripten is likely to be a different cross target (https://github.com/NixOS/nixpkgs/pull/217428), so that wouldn't really be at all aligned with how pkgsCross expects the world to work.  It's not really designed for building for more than one target at the same time, so where we can do that in another way (e.g. by making one Rust toolchain able to build both native and Rust code, for packages like polkadot or lldap) that lets us avoid those limitations of pkgsCross.

Reply:
I don't think Emscripten is particularly related tbh. As I previously mentioned a better example is ghcjs, this target is more like ghcjs, it targets the web, it doesn't have a libc, it's compiler specific (rustc) (maybe TinyGo too?), except ghcjs is much more obtrusive as there a couple of explicit conditionals referencing it in the stdenv logic. The changes in this PR are meant to be somewhat generic and only touch buildRustCrate for now.